src/java.base/share/classes/java/lang/Class.java

```diff
@@ -392,9 +392,10 @@ static String typeVarBounds(TypeVariable<?> typeVar) {
      * caller frame on the stack (e.g. when called directly from a JNI
      * attached thread), the system class loader is used.
      *
-     * @param      className   the fully qualified name of the desired class.
-     * @return     the {@code Class} object for the class with the
-     *             specified name.
+     * @param     className the <a href="#binary-name">binary name</a> of the class
```

Comment:
The new text is probably more correct, but it is less easy to understand. I think "fully qualified name" of a class is a well-known term, but "binary name" is rather unknown.

I suggest changing the text to mention both terms. For example like this:

```suggestion
     * @param     className  the <a href="#binary-name">binary name</a>) of the class, for example its fully qualified name
```

Reply:
Binary name is defined in JLS. Fully qualified name is ambiguous as it may refer to the internal name or a descriptor string.